rpc-server/src/utils.rs

```diff
@@ -94,37 +94,47 @@ pub async fn update_final_block_height_regularly(
 /// `readrpc_response`: a `Result<serde_json::Value, serde_json::Error>` object representing the results from Read RPC.
 /// `client`: `near_jsonrpc_client::JsonRpcClient`.
 /// `params`: `near_jsonrpc_client::methods::RpcMethod` trait.
+///
+/// In case of a successful comparison, the function returns `Ok(())`.
+/// Otherwise, it returns `Err(ShadowDataConsistencyError)`.
 #[cfg(feature = "shadow_data_consistency")]
 pub async fn shadow_compare_results<M>(
     readrpc_response: Result<serde_json::Value, serde_json::Error>,
     client: near_jsonrpc_client::JsonRpcClient,
     params: M,
-) where
+) -> Result<(), ShadowDataConsistencyError>
+where
     M: near_jsonrpc_client::methods::RpcMethod + std::fmt::Debug,
     <M as near_jsonrpc_client::methods::RpcMethod>::Response: serde::ser::Serialize,
     <M as near_jsonrpc_client::methods::RpcMethod>::Error: std::fmt::Debug,
 {
-    tracing::debug!("Compare results. {:?}", params);
+    tracing::debug!(target: "shadow_data_consistency", "Compare results. {:?}", params);
 
     let readrpc_response_json = match readrpc_response {
         Ok(readrpc_response_json) => readrpc_response_json,
         Err(err) => {
-            tracing::error!(target: "is_not_consistency", "Parse hundredx response error: {:#?}", err);
-            return;
+            // TODO: remove this log record after the methods calling function `shadow_compare_results` are refactored
```

Comment:
I've added the `TODO`s like this to smooth the transition since @kobayurii is going to refactor the methods and only after that, we can say we're fully adopted the refactored version.